### PR TITLE
[FIX DB MIGRATIONS] we can't reset db on servers

### PR DESCRIPTION
**Pull request by RuslanHamidullin**
So guys we can't reset databases on servers.
I did some corrects related to latest AwardYear related migrations.
Now it's one migration, which works by following way:
```
class AwardYearsStructureCorrects < ActiveRecord::Migration
  def up
    # Clean up existing Settings
    Settings.destroy_all

    # Add award_year_id to Settings (without null: false)
    remove_column :settings, :year
    add_column :settings, :award_year_id, :integer
    Settings.reset_column_information

    # Create award_years table
    create_table :award_years do |t|
      t.integer :year, null: false
      t.timestamps null: false
    end

    # Populate Award Years.
    # Settings and Deadlines will be created automatically
    AwardYear::AVAILABLE_YEARS.each do |year|
      AwardYear.create!(year: year)
    end

    # # Add null: false and index for award_year_id for Settings
    change_column :settings, :award_year_id, :integer, null: false
    add_index :settings, :award_year_id

    # # Add award_year_id to FormAnswer (without null: false)
    add_column :form_answers, :award_year_id, :integer
    FormAnswer.reset_column_information

    # # Populate award_year_id value for existing FormAnswers
    FormAnswer.all.each do |form_answer|
      award_year_attr = form_answer.attributes["award_year"]
      award_year = AwardYear.find_by_year(award_year_attr)
      form_answer.update_column(:award_year_id, award_year.id)
    end

    # # Remove award_year column from FormAnswer
    remove_column :form_answers, :award_year

    # # # Add null: false and index for award_year_id for FormAnswer
    change_column :form_answers, :award_year_id, :integer, null: false
    add_index :form_answers, :award_year_id
  end
end

```

### CODE REVIEW

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def render_main_header
   end
 
   def render_certificate_info
-    title = "THE QUEEN’S AWARDS FOR ENTERPRISE #{form_answer.award_year}: AUDITOR’S CERTIFICATE"
+    title = "THE QUEEN’S AWARDS FOR ENTERPRISE #{form_answer.award_year.year}: AUDITOR’S CERTIFICATE"
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [101/100]

File: db/migrate/20150410091747_award_years_structure_corrects.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+class AwardYearsStructureCorrects < ActiveRecord::Migration
+  def up
```
Comment (matthewford) [inline db/migrate/20150410091747_award_years_structure_corrects.rb]: Assignment Branch Condition size for up is too high. [23.09/15]

File: spec/unit/pdf_generators/case_summary_pdfs/base_spec.rb [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
+
+  let!(:form_answer_current_year_innovation) do
+    FactoryGirl.create :form_answer, :submitted, :innovation
   end
 
-  let!(:form_answer_2015_innovation) do
+  let!(:form_answer_next_year_innovation) do
     FactoryGirl.create :form_answer, :submitted, :innovation,
-                        award_year: 2015
+                        award_year_id: next_year
```
Comment (matthewford) [inline spec/unit/pdf_generators/case_summary_pdfs/base_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/unit/pdf_generators/case_summary_pdfs/base_spec.rb [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
-                        award_year: 2014
+  let!(:form_answer_current_year_trade) do
+    FactoryGirl.create :form_answer, :submitted, :trade
   end
 
-  let!(:form_answer_2015_trade) do
+  let!(:form_answer_next_year_trade) do
     FactoryGirl.create :form_answer, :submitted, :trade,
-                        award_year: 2015
+                        award_year_id: next_year
```
Comment (matthewford) [inline spec/unit/pdf_generators/case_summary_pdfs/base_spec.rb]: Align the parameters of a method call if they span more than one line.

File: spec/unit/pdf_generators/case_summary_pdfs/base_spec.rb [hunk 5]
```diff
@@ -1,28 +1,30 @@
 require 'rails_helper'
 
 describe "CaseSummaryPdfs::Base" do
-  let!(:form_answer_2014_innovation) do
-    FactoryGirl.create :form_answer, :submitted, :innovation,
-                        award_year: 2014
+  let(:next_year) {
```
Comment (matthewford) [inline spec/unit/pdf_generators/case_summary_pdfs/base_spec.rb]: Avoid using {...} for multi-line blocks.

**Comment by matthewford** [comment]
:+1: prefer not to break migrations

**Comment by RuslanHamidullin** [comment]
Here are 3 failed specs:
```
rspec ./spec/mailers/users/collaboration_mailer_spec.rb:77 
rspec ./spec/mailers/users/collaboration_mailer_spec.rb:43 
rspec ./spec/models/users_import/builder_spec.rb:7
```
I can fix users/collaboration_mailer_spec.rb,
models/users_import/builder_spec.rb
```
UsersImport::Builder#process imports the users
     Failure/Error: expect(row.created_at).to eq(Date.new(2014, 8, 4))
       
       expected: Mon, 04 Aug 2014
            got: 2014-04-08 01:00:00.000000000 +0100
       
       (compared using ==)
       
       Diff:
       @@ -1,2 +1,2 @@
       -Mon, 04 Aug 2014
       +Tue, 08 Apr 2014 01:00:00 BST +01:00
       
     # ./spec/models/users_import/builder_spec.rb:13:in `block (3 levels) in <top (required)>'
```
is better to review by @jaleszek

**Pull request merged**